### PR TITLE
Updated Self-Deployment and Subscriber Manifest

### LINKED ISSUE #1940: Integrate self-deployer with admin project creation in GQL server
- Self-Deployer should only start after admin creates their project
- Need to update the manifests accordingly
- Perform self-cluster registration on GQL-Server

### PR DESCRIPTION
**Pull request by gdsoumya**
Signed-off-by: Soumya Ghosh Dastidar <gdsoumya@gmail.com>

## Proposed changes

- Self-Deployer starts only after admin project is created
- Updated the subscriber manifest, now all the components can be installed in one-go

Fixes #1940 

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by rajdas98** [comment]
In this PR, we don't need to use a dynamic client or self-deployer manifest. As we are known to resource
Here we can use simple client-go and using its functions to create a self-deployer deployment

File: litmus-portal/backend/workflow-agent/pkg/cluster/events/util.go [hunk 1]
```diff
@@ -1,13 +1,14 @@
 package events
 
 import (
+	v1 "Projects/go/pkg/mod/k8s.io/apimachinery@v0.17.9/pkg/apis/meta/v1"
```
Comment (rajdas98) [inline litmus-portal/backend/workflow-agent/pkg/cluster/events/util.go:4]: Why are we using the local pkg?

Comment (gdsoumya) [reply litmus-portal/backend/workflow-agent/pkg/cluster/events/util.go]: Not sure when that changed. Fixed it.

**Comment by gdsoumya** [comment]
> In this PR, we don't need to use a dynamic client or self-deployer manifest. As we are known to that resource i.e., `Deployment`
> Here we can use simple client-go and using its functions to create a self-deployer deployment with its ENV

We can do that, but I believe having it read the data from the yaml is much more readable and easy to maintain as we can just make changes to the yaml directly instead of the code. Also Building the docker image will also be easier if we just make changes to the yaml as everything else will be taken from cache.

**Comment by rajdas98** [comment]
It's a single single deployment, not required to be in separate. If you use simple client go, it will reduce so many extra steps and create deployment object is also readable

**Comment by gdsoumya** [comment]
> It's a single single deployment, not required to be in separate file. If you use simple client go, it will reduce so many extra steps and creating a k8s deployment object is also readable

We will have to update the code directly everytime we change something in the manifest, it might reduce a few steps to do it directly but currently we just have to make 1 function call to apply it.

**Review by rajdas98 — APPROVED** [review]

**Pull request merged**